packages/block-library/src/search/edit.native.js

```diff
@@ -262,6 +298,7 @@ export default function SearchEdit( {
 								setAttributes( { buttonText: html } )
 							}
 							minWidth={ MIN_BUTTON_WIDTH }
+							maxWidth={ blockWidth - MARGINS }
```

Comment:
I did notice one oddity, but I'm not sure what portion of the code to attribute it to. When entering text into the button that causes the button to expand its width, the button resizes to fit the new character a split second _after_ the text is appended. There is a brief moment where the new character is visible within the "padding" of the button. It's not terribly disruptive, but creates a small bit of jumping in the layout. Likely not worth attempting to address in this PR, but I wanted to point it out nonetheless.

https://user-images.githubusercontent.com/438664/114952827-c0738080-9e1c-11eb-9293-5956db70f2e1.mov

Reply:
After looking at this one last time, I'm thinking this oddity is merely the nature of how the `RichText` and Aztec editor is implemented. I noticed the Button block also suffers from this same behavior, so it is not related to the styling of the Search block whatsoever.